public/app/core/components/AppChrome/TopBar/InviteUserButtonUtils.test.tsx

```diff
@@ -0,0 +1,94 @@
+import { AccessControlAction } from 'app/types/accessControl';
+
+import { performInviteUserClick, shouldRenderInviteUserButton } from './InviteUserButtonUtils';
+
+// Mock dependencies
+jest.mock('@grafana/runtime', () => ({
+  ...jest.requireActual('@grafana/runtime'),
+  reportInteraction: jest.fn(),
+  config: {
+    featureToggles: {} as Partial<FeatureToggles>,
+    externalUserMngLinkUrl: '',
+  },
+}));
+
+jest.mock('app/core/core', () => ({
+  contextSrv: {
+    hasPermission: jest.fn(),
+  },
+}));
+
+jest.mock('app/features/users/utils', () => ({
+  getExternalUserMngLinkUrl: jest.fn(),
+}));
+
+const mockReportInteraction = jest.mocked(reportInteraction);
+const mockConfig = jest.mocked(config);
+const mockContextSrv = jest.mocked(contextSrv);
+const mockGetExternalUserMngLinkUrl = jest.mocked(getExternalUserMngLinkUrl);
+
+// Type assertion to make mockConfig.featureToggles assignable
+const mockFeatureToggles = mockConfig.featureToggles as Partial<FeatureToggles>;
+
+// Mock window.open
+const mockWindowOpen = jest.fn();
+Object.defineProperty(window, 'open', {
+  value: mockWindowOpen,
+  writable: true,
+});
+
+describe('InviteUserButtonUtils', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    // Set up default mocks
+    mockFeatureToggles.inviteUserExperimental = true;
+    mockConfig.externalUserMngLinkUrl = 'https://example.com/invite';
+    mockContextSrv.hasPermission.mockReturnValue(true);
+    mockGetExternalUserMngLinkUrl.mockReturnValue('https://example.com/invite?cnt=test');
+  });
+
+  describe('shouldRenderInviteUserButton', () => {
+    it('should return true when all conditions are met', () => {
+      expect(shouldRenderInviteUserButton()).toBe(true);
+      expect(mockContextSrv.hasPermission).toHaveBeenCalledWith(AccessControlAction.OrgUsersAdd);
+    });
+
+    it('should return false when feature toggle is disabled', () => {
+      mockFeatureToggles.inviteUserExperimental = false;
+
+      expect(shouldRenderInviteUserButton()).toBe(false);
+    });
+
+    it('should return false when URL is not configured', () => {
+      mockConfig.externalUserMngLinkUrl = '';
+
+      expect(shouldRenderInviteUserButton()).toBeFalsy();
+    });
+
+    it('should return false when user lacks permission', () => {
+      mockContextSrv.hasPermission.mockReturnValue(false);
+
+      expect(shouldRenderInviteUserButton()).toBe(false);
+    });
+  });
+
+  describe('performInviteUserClick', () => {
+    it('should report interaction, get URL, and open window', () => {
+      const placement = 'top_bar_right';
+      const cnt = 'invite-user-test';
+
+      performInviteUserClick(placement, cnt);
+
+      expect(mockReportInteraction).toHaveBeenCalledWith('invite_user_button_clicked', {
+        placement,
+      });
+      expect(mockGetExternalUserMngLinkUrl).toHaveBeenCalledWith(cnt);
+      expect(mockWindowOpen).toHaveBeenCalledWith('https://example.com/invite?cnt=test', '_blank');
```

Comment:
Can we improve tests to better reflect the expected behavior? For instance, the cnt here is `invite-user-test`, but `mockWindowOopen` asserts a different value `test` because the mock was static. We should mock and assert dynamically using the `cnt` declared on each test.